### PR TITLE
rwedid: init at 0.3.2

### PR DESCRIPTION
**Pull request by Janik-Haag**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 - [GoToSocial](https://gotosocial.org/), an ActivityPub social network server, written in Golang. Available as [services.gotosocial](#opt-services.gotosocial.enable).
 
+- [rwedid](https://codeberg.org/ral/rwedid), A tool to read and write EDID data over an I2C bus. Available as [programs.rwedid](#opt-programs.river.enable).
```
Comment (emilylange) [inline nixos/doc/manual/release-notes/rl-2311.section.md:19]:
```suggestion
- [rwedid](https://codeberg.org/ral/rwedid), A tool to read and write EDID data over an I2C bus. Available as [programs.rwedid](#opt-programs.rwedid.enable).
```

**Comment by yu-re-ka** [comment]
I don't think we need an own NixOS module for this. Users can easily add the package to `services.udev.packages` if they wish this behavior.

File: nixos/modules/hardware/rwedid.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+let
+  cfg = config.programs.rwedid;
+in {
+  meta.maintainers = with lib.maintainers; [ janik ];
+
+  options = {
+    programs.rwedid = {
+      enable = lib.mkEnableOption (lib.mdDoc "Read and write EDID data over an I2C bus");
+      package = lib.mkPackageOptionMD pkgs "rwedid" { };
```
Comment (yu-re-ka) [inline nixos/modules/hardware/rwedid.nix:16]: We don't usually include a `.package` option in NixOS modules, unless there are multiple options available.

**Comment by Janik-Haag** [comment]
I can remove the module if you want me to, It also creates a group. And wireshark basically does the same.

File: pkgs/tools/video/rwedid/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  postInstall = ''
+    mkdir -p $out/etc/udev/rules.d
+    echo 'SUBSYSTEM=="i2c-dev",KERNEL=="i2c-[0-9]*", ATTRS{class}=="0x030000", TAG+="uaccess"' > $out/etc/udev/rules.d/60-ddcutil.rules
+  '';
+
+  meta = with lib; {
+    description = "Read and write EDID data over an I2C bus";
+    homepage = "https://codeberg.org/ral/rwedid";
+    license = licenses.mit;
```
Comment (emilylange) [inline pkgs/tools/video/rwedid/default.nix:38]:
```suggestion
    license = licenses.mit;
    platforms = platforms.linux;
```

Seems to be Linux-only software (as it requires a Linux kernel module).

**Review by emilylange — CHANGES_REQUESTED** [review]

**Comment by emilylange** [comment]
No no @Janik-Haag, please keep the module.
@yu-re-ka suggested removing the `package` option (`package = lib.mkPackageOptionMD pkgs "rwedid" { };`) specifically. Not the module in itself.

I think there is just a misunderstanding here.

**Comment by yu-re-ka** [comment]
> No no @Janik-Haag, please keep the module.

No, I also suggested removing the module, since it basically just creates a shortcut to express two lines in one. See my earlier comment.

If each package with udev rules gets its own module, we would have a lot of modules. Just to name a few: ddcutil, sparrow, mixxx, roomeqwizard, limesuite, uhd.
This would bloat nixos eval a lot, and in our current state each new nixos module increases the eval time of all nixos systems, no matter if they use the module or not.

> And wireshark basically does the same.

No, the wireshark module creates an suid wrapper.
But `adb` would be an example of a module that does exactly this.

But then also these udev rules are not really specific to rwedid. Maybe this specific udev rule should be included unconditionally on all nixos systems, so that users can easily get access to i2c devices by joining a group?

**Comment by emilylange** [comment]
Whoops, sorry, my bad^^

**Comment by Janik-Haag** [comment]
>But then also these udev rules are not really specific to rwedid. Maybe this specific udev rule should be included unconditionally on all nixos systems, so that users can easily get access to i2c devices by joining a group?

I would say I'm not experienced enough with i2c to judge that, I just copied the ddcutil i2c udev rules and they worked fine with the hardware. 

> Seems to be Linux-only software (as it requires a Linux kernel module).

Yeah you are right, the kernel module isn't really a problem but it uses linux only rust crates so..

I don't have really have hard feelings about the module, I can also just add a longDescription attribute that explains people they have to add the group and also install the package with `services.udev.packages =with pkgs; [ rwedid ];` that's what I usually tell other people when they have a package that needs some special settings set to work to make it a better user experience is better.

**Comment by Janik-Haag** [comment]
what is keeping this from getting merged?

**Review by emilylange — APPROVED** [review]

**Pull request merged**